packages/expo-cli/src/commands/eas-build/utils/misc.ts

```diff
@@ -46,4 +51,23 @@ async function printBuildResults(builds: (Build | null)[]): Promise<void> {
   }
 }
 
-export { printLogsUrls, printBuildResults };
+function printDeprecationWarnings(deprecationInfo?: DeprecationInfo): void {
+  if (!deprecationInfo) {
+    return;
+  }
+  if (deprecationInfo.type === 'internal') {
+    log.warn('This command is using API that soon will be deprecated, please update expo-cli.');
+    log.warn("Changes won't affect your project confiuration.");
+    log.warn(deprecationInfo.message);
+  } else if (deprecationInfo.type === 'user-facing') {
+    log.warn('This command is using API that soon will be deprecated, please update expo-cli.');
+    log.warn(
+      'There might be some changes necessary to your project configuration, latest expo-cli will provide more specific error messages.'
+    );
+    log.warn(deprecationInfo.message);
+  } else {
+    log(deprecationInfo); // should not happen
```

Comment:
add a warning with `An unexpected warning was encountered. Please report it as a bug:` this matches the style of our other unexpected errors.